### PR TITLE
AppUtil: Fix read safemem pointer

### PR DESCRIPTION
**Pull request by EXtremeExploit**
Earlier it was writing the buffer pointer pointer, now its the actual buffer pointer where data is located

### CODE REVIEW

File: vita3k/modules/SceAppUtil/SceAppUtil.cpp [hunk 1]
```diff
@@ -363,7 +363,7 @@ static SceInt32 SafeMemory(EmuEnvState &emuenv, const void *buf, SceSize bufSize
         write_file(fd, safe_mem.data(), SCE_APPUTIL_SAFEMEMORY_MEMORY_SIZE, emuenv.io, export_name);
         close_file(emuenv.io, fd, export_name);
     } else
-        memcpy(&buf, &safe_mem[offset], bufSize);
+        memcpy(const_cast<void *>(buf), &safe_mem[offset], bufSize);
```
Comment (Macdu) [inline vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: Can you remove the `const_cast` and instead remove the `const ` in the function declaration?

Comment (Zangetsu38) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: no is const, keep it

Comment (Macdu) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: If you need to const cast, then it's not const, as you can see this function is writing to the location this pointer points to.

Comment (Zangetsu38) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: save is const, load, i hgave code like that for keep const in save

Comment (Macdu) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: Yes, save is const but load is not and this function does both so do not make it const. `const_cast` should only be used really exceptionally.

Comment (sunho) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: it should be changed on function declaration as macdu said. Gotta remote const_cast.

Comment (Zangetsu38) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: > it should be changed on function declaration as macdu said. Gotta remote const_cast.

what function declared ? don't touch on header of official function please 💀

Comment (Macdu) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: We are not talking about the header of the official functions which are correct, we are talking about the static function `SafeMemory`...

Comment (Zangetsu38) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: noworks for me, remove it get crash on load save in escape plain

Comment (Zangetsu38) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: Escape plain using Save mode for load save, and after in everyu chapter finished, no works without const

Comment (EXtremeExploit) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: sceAppUtilSaveSafeMemory has a const void* so i have to use const_cast there or not make the buffer not const on that function, which one is prefered?

Comment (Zangetsu38) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: Const cast is good for me, is not possible remove const for save

Comment (Macdu) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp:366]: Yes, use the `const_cast` in `sceAppUtilSaveSafeMemory`, it is better.

Comment (EXtremeExploit) [reply vita3k/modules/SceAppUtil/SceAppUtil.cpp]: should be good now

**Review by Macdu — APPROVED** [review]

**Pull request merged**